### PR TITLE
Work-in-progress related to phoronix test suite

### PR DESCRIPTION
**Pull request by sarahn**
Signed-off-by: Sarah Newman <sarah.newman@computer.org>

### CODE REVIEW

**Review by mydatascience — COMMENTED** [review]
As far as I get. - this is to test the linux itself. It has io and some workload testing. Is it good for Eve itself? Eve is mostly running containers or VMs. Running python looks not relevant? Is it also relevant to run compression?

**Comment by rvs** [comment]
This is to test how well EVE does virtualization basically. Thus you need benchmarks that run *inside* the VMs (both Linux and Windows) and that's where Phoronix comes in. We need to run these benchmarks periodically (on selected types of hardware) and make sure that EVE virtualization performance doesn't decrease. We will also need these benchmarks when we make decision on what implementation to chose (e.g. swapping ext4 for zfs for example, etc.).

Makes sense @mydatascience ?

**Comment by mydatascience** [comment]
Yes, running in container and will add to jenkins builds

**Comment by sarahn** [comment]
I put a windows image publicly at https://hub.docker.com/r/sarahnp/win-phoronix It's like the EVE images where the image is put on stdout if you run it.

File: experimental/benchmark/phoronix/scripts/windows-runner.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+set -o pipefail
+
+DIR="$(readlink -f "$(dirname "$0")")"
+
+set -x
+
+#PORT=8022
+#HOST=127.0.0.1
+#KEY="${DIR}/../id_rsa"
```
Comment (giggsoff) [inline experimental/benchmark/phoronix/scripts/windows-runner.sh:13]: Hi, @sarahn. Can you please put private part of `phoronix.pub` somewhere? Or we need to use some kind of cloud-init iso?
We can use `Passw0rd!`, but this variant is interactive.

Also, I try to use `ssh-copy-id` without success:
```
giggsoff@giggsoff-PC:/data/sarah$ ssh-copy-id -i ~/.ssh/id_rsa -p 8022 IEUser@127.0.0.1
/usr/bin/ssh-copy-id: INFO: Source of key(s) to be installed: "/home/giggsoff/.ssh/id_rsa.pub"
/usr/bin/ssh-copy-id: INFO: attempting to log in with the new key(s), to filter out any that are already installed
/usr/bin/ssh-copy-id: INFO: 1 key(s) remain to be installed -- if you are prompted now it is to install the new keys
IEUser@127.0.0.1's password: 
exec : The term 'exec' is not recognized as the name of a cmdlet, function, script file, or operable program. Check 
the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ exec sh -c 'cd ; umask 077 ; mkdir -p .ssh && { [ -z `tail -1c .ssh/a ...
+ ~~~~
    + CategoryInfo          : ObjectNotFound: (exec:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

File: experimental/benchmark/phoronix/images/ansible/roles/win-phoronix/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  * ansible
+  * python3-winrm
+
+### Manual setup in Windows VM
+
+  * Log in using default credential IEUser / Passw0rd!
+  * Click on network connection
+  * Click on "network and internet settings"
+  * Click on "change connection properties"
+  * Select "private" not "public"
```
Comment (giggsoff) [inline experimental/benchmark/phoronix/images/ansible/roles/win-phoronix/README.md:70]: We need to do it manually? Can we do it via ssh or something else?
Now, I get:
```
pts/fio-1.13.2:
        Test Installation 1 of 1
        1 File Needed [3.55 MB]
        Downloading: fio-3.18-x64.zip                                                                       [3.55MB]
            Download Failed: http://bsdio.com/fio/releases/fio-3.18-x64.zip
            Downloading of needed test files failed.

```

File: experimental/benchmark/phoronix/scripts/windows-runner.sh (hunk 1 above)
Comment (sarahn) [reply experimental/benchmark/phoronix/scripts/windows-runner.sh:13]: The playbook produces an image that runs cloud-init at next boot. I assumed you would use cloud-init to add keys to the IEUser user.  If you don't have cloud-init, it's OK to just make a new key pair and modify the playbook with the new key pair.

Comment (sarahn) [reply experimental/benchmark/phoronix/scripts/windows-runner.sh:13]: I'm not surprised ssh-copy-id does not work. The shell in the windows VM is powershell, not bash.

File: experimental/benchmark/phoronix/images/ansible/roles/win-phoronix/README.md (hunk 2 above)
Comment (sarahn) [reply experimental/benchmark/phoronix/images/ansible/roles/win-phoronix/README.md:70]: > We need to do it manually? Can we do it via ssh or something else?

Yes and no. The reason to perform this manual step is to get WinRM https://docs.microsoft.com/en-us/windows/win32/winrm/portal to listen on the network interface, which is the service you need to automate everything else, including installing SSH. SSH is not installed at this step, that happens in the ansible playbook.

You might be able to get around this requirement by booting the VM in VirtualBox or a different image where the virtualization technology matches what MS had, if that means that the image comes up with an existing interface already on a private network.

> Now, I get:
> 
> ```
> pts/fio-1.13.2:
>         Test Installation 1 of 1
>         1 File Needed [3.55 MB]
>         Downloading: fio-3.18-x64.zip                                                                       [3.55MB]
>             Download Failed: http://bsdio.com/fio/releases/fio-3.18-x64.zip
>             Downloading of needed test files failed.
> ```

I don't know, it used to work. I might try starting by seeing if there's a newer phoronix version.

File: experimental/benchmark/phoronix/scripts/windows-runner.sh (hunk 1 above)
Comment (sarahn) [reply experimental/benchmark/phoronix/scripts/windows-runner.sh:13]: I can email you the private key if that's what you really need but the existing key was mostly there for me to debug if cloud-init wasn't working.

**Comment by mydatascience** [comment]
Plan to close  this PR. Will wait  a while if someone still wanna transform it to something else.

**Comment by uncleDecart** [comment]
@giggsoff can we close this issue?

**Pull request closed without merging**